tools/src/main/java/com/nedap/archie/flattener/RulesFlattener.java

```diff
@@ -69,12 +69,17 @@ private int findRuleWithTag(RulesSection rules, String tag) {
         return -1;
     }
 
+
     private void addVariableAndTagPrefix(RuleStatement clonedStatement, String variablePrefix, String tagPrefix, String pathPrefix) {
         if(clonedStatement instanceof Assertion) {
             Assertion assertion = (Assertion) clonedStatement;
             if(!Strings.isNullOrEmpty(assertion.getTag())) {
                 assertion.setTag(tagPrefix + assertion.getTag());
             }
+            if (!(assertion.getExpression() instanceof ForAllStatement)) {
+                setVariableReferencePrefix(assertion.getExpression(),"item");
```

Comment:
So you add item, then item will automatically get prefixed? Does that mean the variable name is always unique?

You also add an empty ModelReference, I assume it will have a prefix added automatically in the next step?

If so, maybe add some comment? This is not intuitively clear from reading this code

Reply:
Added a comment.